js/bitmart.js

```diff
@@ -1816,9 +1818,16 @@ module.exports = class bitmart extends Exchange {
         if (ioc) {
             request['type'] = 'ioc';
         }
-        const response = await this[method] (this.extend (request, params));
+        const [ marginMode, query ] = this.handleMarginModeAndParams ('createOrder', params);
+        if ((marginMode === 'cross') || (marginMode === 'isolated')) {
+            if (marginMode !== 'isolated') {
+                throw new NotSupported (this.id + ' createOrder() is only available for isolated margin');
+            }
+            method = 'privatePostSpotV1MarginSubmitOrder';
+        }
+        const response = await this[method] (this.extend (query, params));
```

Comment:
@Dan-krm this is crashing because by doing this `this.extend (query, params)` you're restoring the "marginMode" parameter and sending it to the exchange.

we just need to send `query` here, which is the result of `params` with `marginMode`omitted.


![image](https://user-images.githubusercontent.com/43336371/182920920-ad84a46e-95fa-49b3-97b6-edaa9c359035.png)

Reply:
@Dan-krm, sorry I might not have been very clear here, but what I meant is that we should extend `request` with `query` only, not `query` and `params`. By sending only `query` we're throwing out the entire request object.

As I said, I did not express it clearly, but I would like to ask that every time you change something in the code, test it again regardless of how tiny the change is, for instance, the method would not work if it was merged as is. 

I've fixed it now, so you don't need to change anything else.